services/proxy/pkg/middleware/oidc_auth.go

```diff
@@ -111,20 +134,57 @@ func (m oidcAuth) getClaims(token string, req *http.Request) (claims map[string]
 	return
 }
 
-// extractExpiration currently just returns a hardcoded default for now. It was
-// supposed to parse and extract the expiration time from the provided
-// access_token.
-// As the access_token is defined as an opaque string. Validating and parsing it
-// can be tricky:
-// 1. Try to treat it as a JWT:
-//    - Verifying the validity of the token requires downloading the propoer public
-//      key from the IDP (uri in "jwks_uri" in ".well-known/openid-configuration"
-// 2. Verify and extract it via the introspection endpoint of the IDP (RFC7662) for
-//    IDPs that provide that feature
-// 3. Other IDP implementation specific methods.
-// 4. Fallback to default value
-func (m oidcAuth) extractExpiration(token string) time.Time {
+func (m oidcAuth) verifyAccessToken(token string) (jwt.RegisteredClaims, error) {
+	switch m.accessTokenVerifyMethod {
+	case config.AccessTokenVerificationJWT:
+		return m.verifyAccessTokenJWT(token)
+	case config.AccessTokenVerificationNone:
+		m.logger.Debug().Msg("Access Token verification disabled")
+		return jwt.RegisteredClaims{}, nil
+	}
+	m.logger.Error().Str("access_token_verify_method", m.accessTokenVerifyMethod).Msg("Unknown Access Token verification setting")
+	return jwt.RegisteredClaims{}, errors.New("Unknown Access Token Verification method")
+}
+
+// verifyAccessTokenJWT tries to parse and verify the access token as a JWT.
+func (m oidcAuth) verifyAccessTokenJWT(token string) (jwt.RegisteredClaims, error) {
+	var claims jwt.RegisteredClaims
+	jwks := m.getKeyfunc()
+	if jwks == nil {
+		return claims, errors.New("Error initializing jwks keyfunc")
+	}
+
+	_, err := jwt.ParseWithClaims(token, &claims, jwks.Keyfunc)
+	m.logger.Debug().Interface("access token", &claims).Msg("parsed access token")
+	if err != nil {
+		var jerr *jwt.ValidationError
+		if errors.As(err, &jerr) && jerr.Is(jwt.ErrTokenMalformed) {
+			m.logger.Info().Err(err).Msg("The access token is not a valid JWT.")
+		} else {
+			m.logger.Info().Err(err).Msg("Failed to parse/verify the access token.")
+		}
```

Comment:
Do we need 2 different messages? Imo the `err` should already contain all the information we need.

Reply:
Fixed